cli/core/pkg/artifact/local.go

```diff
@@ -45,3 +45,27 @@ func (l *LocalArtifact) Fetch() ([]byte, error) {
 	}
 	return b, nil
 }
+
+// FetchTest reads test plugin artifact based on the local plugin artifact path
+// To fetch the test binary from the plugin we are using plugin binary path and creating test plugin path from it
+// If the plugin binary path is `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/tanzu-cluster-darwin_amd64`
+// then test plugin binary will be under `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/test/` directory
+func (l *LocalArtifact) FetchTest() ([]byte, error) {
+	// test plugin directory based on the plugin binary location
+	testPluginDir := filepath.Join(filepath.Dir(l.Path), "test")
+	dirEntries, err := os.ReadDir(testPluginDir)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to read test plugin directory")
+	}
+	if len(dirEntries) != 1 {
+		return nil, errors.Errorf("unable to find test plugin under test directory(%v). Found %v files under that directory", testPluginDir, len(dirEntries))
+	}
+
+	// Assuming the file under the test directory is the test plugin binary
```

Comment:
Q: Is there any way of verifying that the file under the test directory is the test plugin binary?

Reply:
I was thinking about this and one thing that came into my mind was to run `<plugin-binary> info` command and check the `group` information from the output. But I wasn't sure about the how important is this validation here considering if it is incorrect we will see the error with `tanzu test plugin <plugin-name>` command invocation. It would be helpful if you can provide your thoughts on this.

```
{"name":"cluster-test","description":"test for cluster","version":"v0.0.1","buildSHA":"","digest":"","group":"Test","docURL":"","completionType":0,"aliases":["cluster-alias"],"installationPath":"","testPluginInstallationPath":"","discovery":"","scope":"","status":"","discoveredRecommendedVersion":""}
```

Reply:
Thanks for the explanation! I'd be ok with`tanzu test plugin <plugin-name>` throwing a meaningful error in order to verify the binary.